olive/passes/onnx/quantization.py

```diff
@@ -377,8 +377,9 @@ def _run_for_config(
         # reload the model and save to output_model_path using the external data config
         # TODO: don't default to use_external_data_format=True if the loading and saving model makes
         # the pass inefficient
-        tmp_dir = tempfile.TemporaryDirectory(prefix="olive_tmp")
```

Comment:
We are using `tmp_dir` and not `self.tmp_dir` since we want the contents in `self.tmp_dir` to persist. It is used to store the `preprocessed` model created at line 326. So that the same input model doesn't need to be preprocessed at every run. 

So this is intentional.

Reply:
Fixed.

Reply:
I don't think the issue here is that there are two tmp_dirs. It's the clean_up of `self.tmp_dir1` since that deletes the original preprocessed model. So the preprocessing would run again in the next run with the same input model.

Reply:
No, it's doesn't anymore because the preprocessed model is saved correctly to the output folder (instead of the working directory as it was earlier). All these path related issues were because of the incompatible `Path` argument to the `quant_pre_process` function. You can try running it locally to verify.

Reply:
Sorry, I mean that line 419 `self.tmp_dir1.cleanup()` would delete the preprocessed model that was saved in the current run. 
So in the next run of this pass with the same input model (during search), line `332` returns `false` and it runs the preprocessing again. 
if we won't do the cleanup for tmp_dir1 every run, then we can reuse the preprocessed model in subsequent runs of the same pass.

Reply:
Personally, I think we can just keep the `tmp_dir` related stuff as before. The `self.tmp_dir` is used for storing persistent files for the object. but the `tmp_dir` inside is used for actually temporary files.

Reply:
Hmmm, removed the delete and added a note for future.

Reply:
I ran into both the issues (disk space and files getting deleted by OS) and that's what landed me around this code. I am working with a very large model and the data file is around 6GB and multiple copies adds up really fast.

Reply:
I will create a work item to track this. The tmp_dirs behave as expected on Linux. On windows, I have seen files getting deleted even though the reference to the tmp_dir still exists (so it should not have been deleted). 
We need to do some investigation into temp directory behaviors on different OS and come up with best practices. Maybe using context managers instead of object references when possible.